src/Server.zig

```diff
@@ -1050,15 +1050,31 @@ fn resolveConfiguration(server: *Server, config_arena: std.mem.Allocator, config
         try std.fs.cwd().makePath(config.global_cache_path.?);
     }
 
-    if (config.build_runner_path == null) blk: {
-        if (config.global_cache_path == null) break :blk;
+    if (config.build_runner_path == null and config.global_cache_path != null) {
+        const build_runner_version = BuildRunnerVersion.selectBuildRunnerVersion(server.runtime_zig_version.?.version);
+
+        const build_runner_file_name = try std.fmt.allocPrint(config_arena, "build_runner_{s}.zig", .{@tagName(build_runner_version)});
+        const build_runner_path = try std.fs.path.resolve(config_arena, &[_][]const u8{ config.global_cache_path.?, build_runner_file_name });
+
+        const build_runner_file = try std.fs.createFileAbsolute(build_runner_path, .{});
+        defer build_runner_file.close();
+
+        const build_config_path = try std.fs.path.resolve(config_arena, &[_][]const u8{ config.global_cache_path.?, "BuildConfig.zig" });
 
-        config.build_runner_path = try std.fs.path.resolve(config_arena, &[_][]const u8{ config.global_cache_path.?, "build_runner.zig" });
+        const build_config_file = try std.fs.createFileAbsolute(build_config_path, .{});
+        defer build_config_file.close();
+
+        try build_config_file.writeAll(@embedFile("build_runner/BuildConfig.zig"));
+
+        try build_runner_file.writeAll(
+            switch (build_runner_version) {
+                inline else => |tag| @embedFile("build_runner/" ++ @tagName(tag) ++ ".zig"),
+            },
+        );
 
-        const file = try std.fs.createFileAbsolute(config.build_runner_path.?, .{});
-        defer file.close();
+        log.debug("using build runner version: {s}", .{@tagName(build_runner_version)});
```

Comment:
I think its better to move this log message somewhere [here](https://github.com/zigtools/zls/blob/9c1f5c8163d64128f032d302dce881ae381d8aa8/src/Server.zig#L867-L889) for consistency.